constellation-app/constellation
# Fix performance and other misc issues with timeline view

### Prerequisites


    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change


Fixed performance issue with timeline view, which would previously cause the timeline view to take a long time to function for large graphs.
Other misc issues were fixed including:

- When nothing is selected, the range of the timeline view would change to be in the millions
- if range is 0, the timeline view can't scroll out or in anymore
- Transactions with no date time are now ignored, before they were treated as having a datetime of 0

### Alternate Designs


N/A

### Why Should This Be In Core?


N/A

### Benefits


Performance boost and misc bug fixes to timeline view.

### Possible Drawbacks


None

### Verification Process



1. Open Consty
2. Create new graph
3. Experimental -> Build Graph -> Structured Graph Builder -> Click "Build"
4. After graph has finished building, Views -> Timeline
5. Observe Timeline View updates comparatively faster than on master branch (or older version of consty)
6. Select everything on the graph (Ctrl + A) and deselect everything (ESC) multiple times in quick succession
7. Observe Consty does not experience any out of memory issues

### Applicable Issues


https://github.com/constellation-app/constellation/issues/2163
